Dockerfile

```diff
@@ -1,3 +1,19 @@
+####################################################################################################
+# ui-builder
+####################################################################################################
+FROM --platform=$BUILDPLATFORM docker.io/library/node:20.12.1 AS ui-builder
+
+RUN npm install --global pnpm
+WORKDIR /ui
+COPY ["ui/package.json", "ui/pnpm-lock.yaml", "./"]
+ARG VERSION
+
+RUN pnpm install
+
+COPY ["ui/", "."]
+
+RUN NODE_ENV='production' VERSION=${VERSION} pnpm run build
+
```

Comment:
This is moved here, unaltered from a few lines down. This stage needs to precede the back-end build now.